lib/grandpa/grandpa.go

```diff
@@ -50,6 +50,39 @@ func NewService(blockState BlockState, voters []*Voter) (*Service, error) {
 	}, nil
 }
 
+func (s *Service) isCompletable() (bool, error) {
+	votes := s.getVotes()
+	prevoted, err := s.getPreVotedBlock()
+	if err != nil {
+		return false, err
+	}
+
+	for _, v := range votes {
+		// check if the current block is a descendant of prevoted block
+		isDescendant, err := s.blockState.IsDescendantOf(prevoted.hash, v.hash)
+		if err != nil {
+			return false, err
+		}
+
+		if !isDescendant {
+			continue
+		}
+
+		// if it's a descendant, check if has >=2/3 votes
```

Comment:
might be nice to break this comment up so that the second part goes with the code block below, for example:

line 71 `get total vote count`
line 77 `check if total vote count >=2/3 votes`